utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -328,10 +340,116 @@ where
+where
+	Block: BlockT + serde::de::DeserializeOwned,
+	Block::Hash: FromStr,
+	<Block::Hash as FromStr>::Err: Debug,
+	NumberFor<Block>: FromStr,
+	<NumberFor<Block> as FromStr>::Err: Debug,
+	ExecDispatch: NativeExecutionDispatch + 'static,
+{
+	let wasm_method = shared.wasm_method;
+	let execution = shared.execution;
+	let heap_pages = if shared.heap_pages.is_some() {
+		shared.heap_pages
+	} else {
+		config.default_heap_pages
+	};
+
+	let mut changes = Default::default();
+	let max_runtime_instances = config.max_runtime_instances;
+	let executor = NativeExecutor::<ExecDispatch>::new(
+		wasm_method.into(),
+		heap_pages,
+		max_runtime_instances,
+	);
+
+	let (mode, block) = match command.state {
+		State::Snap{ snapshot_path, url } => {
+			// `--header-at` is a required option for `snap`
+			let header_hash: Block::Hash = command.header_at
+				.expect("`--header-at` is a required option for `execute-block snap`.")
+				.parse()
+				.map_err(|e| format!("Could not parse header hash: {:?}", e))?;
+			let block: Block = rpc_api::get_block::<Block, _>(url.clone(), header_hash).await?;
+
+			let mode = Mode::Offline(OfflineConfig {
+				state_snapshot: SnapshotConfig::new(snapshot_path),
+			});
+
+			(mode, block)
+		},
+		State::Live { url, snapshot_path, block_at, modules } => {
+			let block_hash: Block::Hash = block_at
+				.ok_or("execute-block requires `block_at` option")?
+				.parse()
+				.map_err(|e| format!("Could not parse hash: {:?}", e))?;
+
+			let block: Block = rpc_api::get_block::<Block, _>(url.clone(), block_hash).await?;
+			let parent_hash = block.header().parent_hash();
+
+			let mode = Mode::Online(OnlineConfig {
+				transport: url.to_owned().into(),
+				state_snapshot: snapshot_path.as_ref().map(SnapshotConfig::new),
+				modules: modules.to_owned().unwrap_or_default(),
+				at: Some(parent_hash.to_owned()),
+				..Default::default()
+			});
+
+			(mode, block)
+		}
+	};
+
+	let ext = {
+		let builder = Builder::<Block>::new().mode(mode);
+		// We do not allow overwriting code because we need exact state for storage root match.
+		let mut ext = builder.build().await?;
+
+		// register externality extensions in order to provide host interface for OCW to the runtime.
+		let (offchain, _offchain_state) = TestOffchainExt::new();
+		let (pool, _pool_state) = TestTransactionPoolExt::new();
+		ext.register_extension(OffchainDbExt::new(offchain.clone()));
+		ext.register_extension(OffchainWorkerExt::new(offchain));
+		ext.register_extension(KeystoreExt(Arc::new(KeyStore::new())));
+		ext.register_extension(TransactionPoolExt::new(pool));
+
+		ext
+	};
+
+	// A digest item gets added when the runtime is processing the block, so we need to pop
+	// the last one to be consistent with what a gossiped block would contain.
+	let (mut header, extrinsics) = block.deconstruct();
+	header.digest_mut().pop();
+	let block = Block::new(header, extrinsics);
+
+	let _ = StateMachine::<_, _, NumberFor<Block>, _>::new(
```

Comment:
`let _ = ` feels like an antipattern in normal use. Is this something we could we could replace with a pattern like

```rust
let _something = { ... };
debug_assert!(_something.has_a_good_property());
```